src/Microsoft.Graph/Models/Extensions/DateTimeZoneExtensions.cs

```diff
@@ -9,27 +9,31 @@ namespace Microsoft.Graph.Extensions
     /// <summary>
     /// Implements Content Converter Extensions
     /// </summary>
-    public static class ContentConverterExtensions
+    public static class DateTimeZoneExtensions
     {
         /// <summary>
         /// Converts DateTimeTimeZone which is a Complex Type to DateTime
         /// </summary>
         /// <param name="dateTimeTimeZone"></param>
+        /// <param name="dateFormat"></param>
+        /// <param name="provider">provide CultureInfo.InvariantCulture as the default</param>
         /// <returns></returns>
-        public static DateTime ToDateTime(this DateTimeTimeZone dateTimeTimeZone)
+        public static DateTime ToDateTime(this DateTimeTimeZone dateTimeTimeZone, string dateFormat, IFormatProvider provider)
         {
-            DateTime dateTime = DateTime.Parse(dateTimeTimeZone.DateTime);
+            DateTime dateTime = DateTime.ParseExact(dateTimeTimeZone.DateTime, dateFormat, provider);
```

Comment:
You are missing the `CultureInfo.InvariantCulture` argument and other functionality that Matt provided in his sample implementation. Please re-read Matt's email and the attachment he provided.